alacritty/src/input.rs

```diff
@@ -188,18 +187,59 @@ impl<T: EventListener> Execute<T> for Action {
             Action::IncreaseFontSize => ctx.change_font_size(FONT_SIZE_STEP),
             Action::DecreaseFontSize => ctx.change_font_size(FONT_SIZE_STEP * -1.),
             Action::ResetFontSize => ctx.reset_font_size(),
-            Action::ScrollPageUp => ctx.scroll(Scroll::PageUp),
-            Action::ScrollPageDown => ctx.scroll(Scroll::PageDown),
+            Action::ScrollPageUp => {
+                // Move keyboard cursor
+                let term = ctx.terminal_mut();
+                let scroll_lines = term.grid().num_lines().0 as isize;
+                term.keyboard_cursor = term.keyboard_cursor.scroll(term, scroll_lines);
+
+                ctx.scroll(Scroll::PageUp);
+            },
+            Action::ScrollPageDown => {
+                // Move keyboard cursor
+                let term = ctx.terminal_mut();
+                let scroll_lines = -(term.grid().num_lines().0 as isize);
+                term.keyboard_cursor = term.keyboard_cursor.scroll(term, scroll_lines);
+
+                ctx.scroll(Scroll::PageDown);
+            },
             Action::ScrollHalfPageUp => {
-                let num_lines = ctx.terminal().grid().num_lines().0 as isize;
-                ctx.scroll(Scroll::Lines(num_lines / 2));
+                // Move keyboard cursor
+                let term = ctx.terminal_mut();
+                let scroll_lines = term.grid().num_lines().0 as isize / 2;
+                term.keyboard_cursor = term.keyboard_cursor.scroll(term, scroll_lines);
+
+                ctx.scroll(Scroll::Lines(scroll_lines));
             },
             Action::ScrollHalfPageDown => {
-                let num_lines = ctx.terminal().grid().num_lines().0 as isize;
-                ctx.scroll(Scroll::Lines(-num_lines / 2));
+                // Move keyboard cursor
+                let term = ctx.terminal_mut();
+                let scroll_lines = -(term.grid().num_lines().0 as isize / 2);
+                term.keyboard_cursor = term.keyboard_cursor.scroll(term, scroll_lines);
```

Comment:
I don't really like that you're updating keyboard_cursor outside of its mode, I know that it won't change anything though.

Reply:
I just think that it's better to keep it short here, rather than checking for which mode we're in for pretty much no purpose.

I'm trying to keep the logic in the action method to an absolute minimum, this is already getting kinda big.